src/main/java/ratpack/zipkin/internal/ZipkinHttpClientImpl.java

```diff
@@ -80,26 +80,25 @@ public void close() {
 
     @Override
     public Promise<ReceivedResponse> request(URI uri, Action<? super RequestSpec> action) {
-        AtomicReference<Span> span = new AtomicReference<>();
+        final AtomicReference<Span> span = new AtomicReference<>();
+
         return delegate
-            .request(uri, action.append(requestSpec -> {
-                WrappedRequestSpec captor = new WrappedRequestSpec(this.handler, this.injector, requestSpec, span);
-                action.execute(captor);
-            }))
+            .request(uri, (RequestSpec requestSpec) -> action.execute(new WrappedRequestSpec(handler, injector, requestSpec, span)))
             .wiretap(response -> responseWithSpan(response, span));
     }
 
     @Override
     public Promise<StreamedResponse> requestStream(URI uri, Action<? super RequestSpec> action) {
-        AtomicReference<Span> span = new AtomicReference<>();
+        final AtomicReference<Span> span = new AtomicReference<>();
+
         return delegate
-            .requestStream(uri, action.append(requestSpec -> {
-                WrappedRequestSpec captor = new WrappedRequestSpec(this.handler, this.injector, requestSpec, span);
+            .requestStream(uri, (RequestSpec requestSpec) -> {
```

Comment:
Looking at this now this makes more sense to me.
  
For some reason before I had it in my head that I shouldn't modify the original action, and I needed to add mine to execute in addition; believing the dual execution needed to happen.